File: Labs/Lab1/ManageVolunteersNativeApplicationKotlin/app/src/main/java/com/example/py7/notesapp/DeleteActivity.java
package com.example.py7.notesapp;

import android.content.ContentValues;
import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;
import android.support.v7.app.AlertDialog;

public class DeleteActivity extends AppCompatActivity {
    DBHelper helper;
    TextView TxTitle, TxDetail;
    long id;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_delete);

        helper = new DBHelper(this);

        id = getIntent().getLongExtra(DBHelper.row_id, 0);

        TxTitle = (TextView)findViewById(R.id.listTitle);
        TxDetail = (TextView)findViewById(R.id.listDetail);

        getData();
    }

    private void getData() {
        Cursor cursor = helper.oneData(id);
        if(cursor.moveToFirst()){
            String title = cursor.getString(cursor.getColumnIndex(DBHelper.row_title));
            String detail = cursor.getString(cursor.getColumnIndex(DBHelper.row_note));

            TxTitle.setText(title);
            TxDetail.setText(detail);
        }
    }

    public boolean onCreateOptionsMenu (Menu menu){
        getMenuInflater().inflate(R.menu.delete_menu, menu);
        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item){
        switch (item.getItemId()){
            case R.id.delete_edit:

                AlertDialog.Builder builder = new AlertDialog.Builder(DeleteActivity.this);
                builder.setMessage("This volunteer will be deleted.");
                builder.setCancelable(true);
                builder.setPositiveButton("Delete", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        helper.deleteData(id);
                        Toast.makeText(DeleteActivity.this, "Deleted", Toast.LENGTH_SHORT).show();
                        finish();
                    }
                });
                builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.cancel();
                    }
                });
                AlertDialog alertDialog = builder.create();
                alertDialog.show();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: Labs/Lab1/ManageVolunteersNativeApplicationKotlin/app/src/main/java/com/example/myapplication/DeleteActivity.kt
package com.example.myapplication

import android.content.DialogInterface
import android.os.Bundle

import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity



class DeleteActivity : AppCompatActivity() {
    var helper: DBHelper? = null
    var TxTitle: TextView? = null
    var TxDetail: TextView? = null
    var id: Long = 0
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_delete)
        helper = DBHelper(this)
        id = getIntent().getLongExtra(DBHelper.row_id, 0)
        TxTitle = findViewById(R.id.listTitle) as TextView?
        TxDetail = findViewById(R.id.listDetail) as TextView?
        data
    }

    private val data: Unit
        private get() {
            val cursor = helper!!.oneData(id)
            if (cursor.moveToFirst()) {
                val title = cursor.getString(cursor.getColumnIndex(DBHelper.row_title))
                val detail = cursor.getString(cursor.getColumnIndex(DBHelper.row_note))
                TxTitle!!.text = title
                TxDetail!!.text = detail
            }
        }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        getMenuInflater().inflate(R.menu.delete_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.delete_edit -> {
                val builder: AlertDialog.Builder = AlertDialog.Builder(this@DeleteActivity)
                builder.setMessage("This volunteer will be deleted.")
                builder.setCancelable(true)
                builder.setPositiveButton(
                    "Delete",
                    DialogInterface.OnClickListener { dialog, which ->
                        helper!!.deleteData(id)
                        Toast.makeText(this@DeleteActivity, "Deleted", Toast.LENGTH_SHORT).show()
                        finish()
                    })
                builder.setNegativeButton(
                    "Cancel",
                    DialogInterface.OnClickListener { dialog, which -> dialog.cancel() })
                val alertDialog: AlertDialog = builder.create()
                alertDialog.show()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}